pkg/engine/datasource/graphql_datasource/batch.go

```diff
@@ -0,0 +1,66 @@
+package graphql_datasource
+
+import (
+	"bytes"
+
+	"github.com/buger/jsonparser"
+	"github.com/cespare/xxhash"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/resolve"
+)
+
+var representationPath = []string{"body", "variables", "representations"}
+
+func mergeFederationInputs(inputs ...[]byte) (*resolve.BatchInput, error) {
```

Comment:
not sure if slice of byte slice is a good function signature to perform well at runtime